### ISSUE TITLE
Some odd build output

### ISSUE THREAD

**Issue by V-Kh** [pydata/bottleneck#207]
When I run `make build` I observe some extremely odd output:
[build.log](https://github.com/kwgoodman/bottleneck/files/3123962/build.log)

1) I assume that `aj == aj` warning is just a `NaN` check? 
2) I've never seen such a weird (read - messy) combination of compiler flags. Where does it come from?

**Comment by kwgoodman** [comment]
The only compiler flag added by bottleneck (have a look in setup.py) is `-O2`

Here's my build log:
```
$ make build
python setup.py build_ext --inplace
running build_ext
building 'bottleneck.reduce' extension
creating build
creating build/temp.linux-x86_64-2.7
creating build/temp.linux-x86_64-2.7/bottleneck
creating build/temp.linux-x86_64-2.7/bottleneck/src
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -fPIC -I/usr/local/lib/python2.7/dist-packages/numpy/core/include -I/usr/include/python2.7 -c bottleneck/src/reduce.c -o build/temp.linux-x86_64-2.7/bottleneck/src/reduce.o -O2
creating build/lib.linux-x86_64-2.7
creating build/lib.linux-x86_64-2.7/bottleneck
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wl,-Bsymbolic-functions -Wl,-z,relro -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security build/temp.linux-x86_64-2.7/bottleneck/src/reduce.o -o build/lib.linux-x86_64-2.7/bottleneck/reduce.so
building 'bottleneck.move' extension
creating build/temp.linux-x86_64-2.7/bottleneck/src/move_median
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -fPIC -I/usr/local/lib/python2.7/dist-packages/numpy/core/include -I/usr/include/python2.7 -c bottleneck/src/move.c -o build/temp.linux-x86_64-2.7/bottleneck/src/move.o -O2
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -fPIC -I/usr/local/lib/python2.7/dist-packages/numpy/core/include -I/usr/include/python2.7 -c bottleneck/src/move_median/move_median.c -o build/temp.linux-x86_64-2.7/bottleneck/src/move_median/move_median.o -O2
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wl,-Bsymbolic-functions -Wl,-z,relro -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security build/temp.linux-x86_64-2.7/bottleneck/src/move.o build/temp.linux-x86_64-2.7/bottleneck/src/move_median/move_median.o -o build/lib.linux-x86_64-2.7/bottleneck/move.so
building 'bottleneck.nonreduce' extension
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -fPIC -I/usr/local/lib/python2.7/dist-packages/numpy/core/include -I/usr/include/python2.7 -c bottleneck/src/nonreduce.c -o build/temp.linux-x86_64-2.7/bottleneck/src/nonreduce.o -O2
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wl,-Bsymbolic-functions -Wl,-z,relro -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security build/temp.linux-x86_64-2.7/bottleneck/src/nonreduce.o -o build/lib.linux-x86_64-2.7/bottleneck/nonreduce.so
building 'bottleneck.nonreduce_axis' extension
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -fPIC -I/usr/local/lib/python2.7/dist-packages/numpy/core/include -I/usr/include/python2.7 -c bottleneck/src/nonreduce_axis.c -o build/temp.linux-x86_64-2.7/bottleneck/src/nonreduce_axis.o -O2
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wl,-Bsymbolic-functions -Wl,-z,relro -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security build/temp.linux-x86_64-2.7/bottleneck/src/nonreduce_axis.o -o build/lib.linux-x86_64-2.7/bottleneck/nonreduce_axis.so
copying build/lib.linux-x86_64-2.7/bottleneck/reduce.so -> bottleneck
copying build/lib.linux-x86_64-2.7/bottleneck/move.so -> bottleneck
copying build/lib.linux-x86_64-2.7/bottleneck/nonreduce.so -> bottleneck
copying build/lib.linux-x86_64-2.7/bottleneck/nonreduce_axis.so -> bottleneck```

**Comment by V-Kh** [comment]
This is strange, I tried two different machines, both of them give similar output.

**Comment by qwhelan** [comment]
@V-Kh Thanks for noticing this, closed by #212
